actors/market/src/lib.rs

```diff
@@ -482,6 +483,13 @@ impl Actor {
             .with_context_code(ExitCode::USR_ILLEGAL_ARGUMENT, || {
                 format!("failed to notify deal with proposal cid {}", valid_deal.cid)
             })?;
+
+            emit::deal_published(
+                rt,
+                valid_deal.proposal.client.id().unwrap(),
+                valid_deal.proposal.provider.id().unwrap(),
+                new_deal_ids[i],
```

Comment:
nit: put `let deal_id = new_deal_ids[i]` at the top of the loop (it's used twice). Even better, change the loop to

```rust
for (valid_deal, &deal_id) in valid_deals.iter().zip(&new_deal_ids) {
```